File: PhoneAccessoriesApp/src/api/apiService.js
import axios from 'axios';

const BASE_URL = 'http://localhost:3000'; // Update to your deployed backend URL

export const apiService = {
  async getAccessories() {
    try {
      const response = await axios.get(`${BASE_URL}/accessories`);
      return response.data;
    } catch (error) {
      throw new Error('Échec du chargement des accessoires');
    }
  },

  async getAccessoriesByCategory(category) {
    try {
      const response = await axios.get(`${BASE_URL}/accessories/category/${category}`);
      return response.data;
    } catch (error) {
      throw new Error('Échec du chargement des accessoires par catégorie');
    }
  },

  async searchAccessories(query) {
    try {
      const response = await axios.get(`${BASE_URL}/accessories/search?q=${query}`);
      return response.data;
    } catch (error) {
      throw new Error('Échec de la recherche');
    }
  },

  async placeOrder(items, address) {
    try {
      const response = await axios.post(`${BASE_URL}/orders`, { items, address });
      return response.data;
    } catch (error) {
      throw new Error('Échec de la commande');
    }
  },

  async getOrderHistory() {
    try {
      const response = await axios.get(`${BASE_URL}/orders/history`);
      return response.data;
    } catch (error) {
      throw new Error('Échec de récupération de l\'historique');
    }
  },
};

File: vat_app/lib/services/api_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:vat_app/models/accessory.dart';
import 'package:vat_app/models/cart_item.dart';

class ApiService {
  static const String baseUrl = 'http://localhost:3000';

  static Future<List<Accessory>> getAccessories() async {
    final response = await http.get(Uri.parse('$baseUrl/accessories'));

    if (response.statusCode == 200) {
      List<dynamic> data = json.decode(response.body);
      return data.map((json) => Accessory.fromJson(json)).toList();
    } else {
      throw Exception('Échec du chargement des accessoires');
    }
  }

  static Future<List<Accessory>> getAccessoriesByCategory(String category) async {
    final response = await http.get(Uri.parse('$baseUrl/accessories/category/$category'));

    if (response.statusCode == 200) {
      List<dynamic> data = json.decode(response.body);
      return data.map((json) => Accessory.fromJson(json)).toList();
    } else {
      throw Exception('Échec du chargement des accessoires par catégorie');
    }
  }

  static Future<List<Accessory>> searchAccessories(String query) async {
    final response = await http.get(Uri.parse('$baseUrl/accessories/search?q=$query'));

    if (response.statusCode == 200) {
      List<dynamic> data = json.decode(response.body);
      return data.map((json) => Accessory.fromJson(json)).toList();
    } else {
      throw Exception('Échec de la recherche');
    }
  }

  static Future<Map<String, dynamic>> placeOrder(List<CartItem> items, String address) async {
    final response = await http.post(
      Uri.parse('$baseUrl/orders'),
      headers: <String, String>{
        'Content-Type': 'application/json; charset=UTF-8',
      },
      body: jsonEncode(<String, dynamic>{
        'items': items.map((item) => item.toJson()).toList(),
        'address': address,
      }),
    );

    if (response.statusCode == 201) {
      return json.decode(response.body);
    } else {
      throw Exception('Échec de la commande');
    }
  }

  static Future<List<dynamic>> getOrderHistory() async {
    final response = await http.get(Uri.parse('$baseUrl/orders/history'));

    if (response.statusCode == 200) {
      return json.decode(response.body);
    } else {
      throw Exception('Échec de récupération de l\'historique');
    }
  }
}